lib/CapeMapper/SiteMapper.js

```diff
@@ -31,14 +31,48 @@ class SiteMapper {
     /**
      *  load relevant files from filesystem and map them using the config. nb. this will not work with azure
      *  @param {Object.<string,Buffer|Buffer[]>} source_data. An array of byte streams to import for each dataset. The key is the ID of the dataset.
-     *  @return {Object}
+     *  @return {Array}
      */
     generate(source_data) {
-        let output = { status:"OK", datasets: []};
-        this.datasetMappers.forEach( (dataset_mapper) => {
-           output.datasets.push( dataset_mapper.generate( source_data[ dataset_mapper.config.id ] ));
-        });
-        return output;
+        var output_data;
+        try {
+            var issues = [];
+            var datasets = [];
+            this.datasetMappers.forEach( (dataset_mapper) => {
+                const dataset_output = dataset_mapper.generate( source_data[ dataset_mapper.config.id ] );
+                datasets.push( dataset_output );
+
+                // turn any issues into errors
+                if( dataset_output.unmapped_headings.length > 0 ) {
+                    issues.push( "Dataset "+dataset_mapper.config.id+" has the following unexpected headings: " 
```

Comment:
Should we add spaces around the `+` operator so that it's easier to read the code?